test/shared/src/main/scala-3/zio/test/Macros.scala

```diff
@@ -263,6 +263,9 @@ object SmartAssertMacros {
                             .span($span)
                         }.asExprOf[TestArrow[Any, A]]
                     }
+                  case (None, _) =>
+                    val span = getSpan(tree)
+                    '{ TestArrow.succeed($expr).span($span) }
                   case _ => throw new Error("NO")
```

Comment:
```suggestion
                  case _ => throw new IllegalArgumentException("Unsupported operation in 'assertTrue'\nPlease open an issue: https://github.com/zio/zio/issues/new")
```

Reply:
I just thought maybe it should not be the problem of the end user if they include some strange compare expressions? We can't really predict what kind of expressions are possible here, however, they might appear in the future, maybe we should simply allow all of them.

What I mean is that instead of throwing this strange exception, which appears only in expressions with compare operations, simply do the `'{ TestArrow.succeed($expr).span($span) }` like it is already done when the tree does not match any other patterns (lines 383-385 in the original code)?

Reply:
I am not sure I 100% understand, but I think you're saying the case you added should be a 'catchall'? Even if that's the case, I think it's better to leave a way to report bugs/issues and we can choose to fix/not fix them. `ERROR("NO")` is not descriptive.

Reply:
Since this is used in a macro, why not throw a proper compiler error with the error message suggested by @hearnadam?

```suggestion
                  case _ => scala.compiletime.error("Unsupported operation in 'assertTrue'\nPlease open an issue: https://github.com/zio/zio/issues/new")
```

Or is that not possible?

Reply:
@kyri-petrou, yes it's possible, I have changed the error message to the one suggested by @hearnadam and used the compiler error like you suggested. The only thing I changed is moved this error expression to a separate `def`, since it is reused three times in places where "NO" exception was before.